backend/sites/src/app/services/folio/folioContents.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { FolioContents } from '../../entities/folioContents.entity';
+import { Repository } from 'typeorm';
+import { FolioContentDTO } from '../../dto/folioContent';
+import { plainToInstance } from 'class-transformer';
+
+@Injectable()
+export class FolioContentsService {
+  constructor(
+    @InjectRepository(FolioContents)
+    private folioContentRepository: Repository<FolioContents>,
+  ) {}
+
+
+  async getSiteForFolio(folioId: string): Promise<FolioContents[]> {
+    try {
+      const sitesForFolioArr = await this.folioContentRepository.find({  
+        relations: ['site','folio'],
+        where: { folioId },
+      });
+      return sitesForFolioArr;
+    } catch (error) {
+      throw error;
+    }
+  }
+
+
+  async addFolioContent(inputDTO: FolioContentDTO): Promise<boolean> {
+    const { siteId, folioId } = inputDTO;
+    try {
+      const existingRecord = await this.folioContentRepository.findOne({
+        where: { siteId, folioId },
+      });
+
+      if (existingRecord) {
+        return false;
+      } else {
+        delete inputDTO.id;
+        const folioContent = plainToInstance(FolioContents, inputDTO); 
+         
+        const result = await this.folioContentRepository.save(folioContent);
+
+        if (result) {
+          return true;
+        }
+      }
+    } catch (error) {
+      throw new Error('Failed to folio content');
+    }
+  }
+
+
+  async deleteFolioContent(folioContentId: string): Promise<boolean> {
+    try {
+      if (folioContentId != '') {
+        const result = await this.folioContentRepository.delete({
+          id: folioContentId,
+        });
+        if (result.affected > 0) return true;
+        else return false;
+      }
+
+      return false;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  async deleteSitesInFolio(folioId: string, siteId: string ): Promise<boolean> {
```

Comment:
IMO the return type for this and other methods wherein only Boolean is stated should be `Promise<{ success: boolean, message?: string }` so that we know the exact reason behind the failure. Let me know your thoughts. Ignore message if success: true and have an error message in case success: false(failure)

Reply:
The delete function delete the record if it find the match record where clause other than that there is no reason. So returning true or false which shows deleted or not